StatusEffect/EverLeaking

```diff
@@ -0,0 +1,57 @@
+extends StatusEffectBase
+
+var _toCheck = [
+	character.hasBodypart(BodypartSlot.Penis),
+	character.hasBodypart(BodypartSlot.Vagina),
+	character.hasWombIn(BodypartSlot.Anus)
+]
+
+func _init():
+	id = StatusEffect.EverLeaking
+	isBattleOnly = false
+
+var rng = RandomNumberGenerator.new()
+func processTime(_secondsPassed: int):
+	# this should run every ten seconds
+	turns -= _secondsPassed
+	if(turns <= 0):
+		# gamble if the player goes into heat/rut or not
+		# chance would be 1 in whatever the max value is
+		if rng.randi_range(0, 100) == rng.randi_range(0, 100):
+			pass
+		
+		turns += 10
+
+func getEffectName():
+	return "Ever-Leaking"
+
+func getEffectDesc():
+	var message = "You're too horny. "
+	match _toCheck:
+		[true, false, false]: # has penis
```

Comment:
I'd just prefer if checks.. You even realize that [true, false, false] isn't descriptive at all so you added comments that explain them. Why not just use the actual functions instead of random arrays of boolean flags? x3